blood/README.md

```diff
@@ -1,22 +1,83 @@
+
+For example, if one parent has blood type AO and the other parent has blood type BB, then the child's possible blood types would be AB and OB, depending on which allele is received from each parent. Similarly, if one parent has blood type AO and the other OB, then the child's possible blood types would be AO, OB, AB, and OO.
 
 ## Implementation Details
 
-TODO
+Complete the implementation of `blood.c` at right, such that it creates a family of a specified generation size and assigns blood type alleles to each family member. The highest generation will have alleles assigned randomly to them.
+
+* First, notice that we've created a `struct` named `person`. Each `person` will have two pointers, pointing to the person's parents. Each parent is of type `person` as well, and they're stored in an array named `parents`. Additionally, each `person` will have two alleles, one from each parent, stored in an array named `alleles`. 
+* Notice that there are four helper functions. 
+  * `create_family()` takes as input an integer, allocates memory for the family, and returns a person with a family of that specified number of generations.
+    * For example, `create_family(3)` will return a person with parents and grandparents.  
+  * `print_family()` takes as input a person and a number signifying the generation that the person is in. This function prints the entire family, their generation numbers, and their blood types, as per the sample output. 
+  * `free_family()` takes as input a person and frees the memory associated with this person and the person's family. Remember that all the memory we allocate, we also have to free. 
+  * `random_allele()` returns a random allele between `A`, `B`, and `O`. 
+* Notice that `main()` is already implemented. 
+  * First, a random seed is set, which depends on the time of executing the program. This allows for different random values to be generated when we call functions incorporating randomness. 
+  * A new family with three generations is then created using `create_family(3)`, and it is stored in a person of type `person` and name `p`. 
+  * The family of `p` is then printed with `print_family(p, 0)`.
+  * Finally, the memory associated with this family is freed with `free_family(p)`. 
+* In `create_family()`, your program should create a family with the specified number of generations. 
+  * When creating a family, first allocate memory for a new person. 
+  * Then, if necessary, recursively create the person's parents and use the parents' alleles to assign the child's alleles. Note the rules for assigning alleles described above—the determination of which of the two alleles the child gets from a parent is random.
+  * If it is not necessary to create parents, set the person's parent pointers to `NULL` and randomly assign its alleles. 
```

Comment:
Let's be explicit here and say that this will be the case if the input to the function is `1` (only one generation)